### PR TITLE
Fix: unable to set max environments in projects

### LINKED ISSUE #524: Unable to set max environments in projects
when using the latest terraform provider:
+ provider [registry.terraform.io/env0/env0](http://registry.terraform.io/env0/env0) v1.2.10

I am trying to create a project with the environment maximums set to null.  If I set them to a number >= 1 then the code works, but if I specify 0 or null I receive errors.

Works:

```
 resource "env0_project_policy" "example" {
  project_id                    = data.env0_project.customer-test.id
  number_of_environments        = 1
  number_of_environments_total  = 1
}
```

Fails:

```
resource "env0_project_policy" "example" {
  project_id                    = data.env0_project.customer-test.id
  number_of_environments        = null
  number_of_environments_total  = null
}
```

Error:

```
│ Error: could not update policy: 400 Bad Request: /body/numberOfEnvironments must be >= 1, /body/numberOfEnvironments must be null, /body/numberOfEnvironments must match a schema in anyOf, /body/numberOfEnvironments must be null, /body/numberOfEnvironments must match a schema in anyOf, /body/numberOfEnvironmentsTotal must be >= 1, /body/numberOfEnvironmentsTotal must be null, /body/numberOfEnvironmentsTotal must match a schema in anyOf, /body/numberOfEnvironmentsTotal must be null, /body/numberOfEnvironmentsTotal must match a schema in anyOf
│ 
│   with env0_project_policy.example,
│   on projects.tf line 10, in resource "env0_project_policy" "example":
│   10: resource "env0_project_policy" "example" {
```

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
Fixes #524 

### Solution

Fixed a few issues (see in comments), and tested manually.

### CODE REVIEW

File: client/project_policy.go [hunk 1]
```diff
@@ -13,23 +13,23 @@ type Policy struct {
 	UpdatedBy                   string  `json:"updatedBy"`
 	RunPullRequestPlanDefault   bool    `json:"runPullRequestPlanDefault"`
 	ContinuousDeploymentDefault bool    `json:"continuousDeploymentDefault"`
-	MaxTtl                      *string `json:"maxTtl,omitempty" tfschema:",omitempty"`
-	DefaultTtl                  *string `json:"defaultTtl,omitempty" tfschema:",omitempty"`
+	MaxTtl                      *string `json:"maxTtl" tfschema:",omitempty"`
```
Comment (TomerHeber) [inline client/project_policy.go:16]: unfortunately, the API needs to pass "null" explicitly therefore, I had to remove the omitempty.
This is incorrect behavior in the backend (Ok for PATCH requests but wrong for PUT requests).

File: client/project_policy.go [hunk 2]
```diff
@@ -13,23 +13,23 @@ type Policy struct {
...
+	DefaultTtl                  *string `json:"defaultTtl" tfschema:",omitempty"`
 }
 
 type PolicyUpdatePayload struct {
 	ProjectId                   string `json:"projectId"`
-	NumberOfEnvironments        *int   `json:"numberOfEnvironments,omitempty"`
-	NumberOfEnvironmentsTotal   *int   `json:"numberOfEnvironmentsTotal,omitempty"`
+	NumberOfEnvironments        *int   `json:"numberOfEnvironments"`
```
Comment (TomerHeber) [inline client/project_policy.go:22]: unfortunately, the API needs to pass "null" explicitly therefore, I had to remove the omitempty.
This is incorrect behavior in the backend (Ok for PATCH requests but wrong for PUT requests).

File: client/project_policy.go [hunk 3]
```diff
@@ -19,8 +19,8 @@ type Policy struct {
 
 type PolicyUpdatePayload struct {
 	ProjectId                   string `json:"projectId"`
-	NumberOfEnvironments        *int   `json:"numberOfEnvironments,omitempty"`
-	NumberOfEnvironmentsTotal   *int   `json:"numberOfEnvironmentsTotal,omitempty"`
+	NumberOfEnvironments        *int   `json:"numberOfEnvironments" tfschema:",omitempty"`
```
Comment (TomerHeber) [inline client/project_policy.go:22]: Removed the omitempty for the json.
In general it shouldn't be required, but it seems to be an issue with how the backend works (expects "null").

File: env0/utils.go [hunk 4]
```diff
@@ -50,15 +50,25 @@ func stringInSlice(str string, strs []string) bool {
...
+func reasourceDataGetValue(fieldName string, omitEmpty bool, d *schema.ResourceData) interface{} {
 	dval := d.Get(fieldName)
 	if dval == nil {
 		return nil
 	}
 
-	_, okInt := dval.(int)
+	ival, okInt := dval.(int)
+
+	if omitEmpty && okInt && ival == 0 {
```
Comment (TomerHeber) [inline env0/utils.go:61]: handles omit empty for empty string and or zero int.
Up until now, it wasn't required. The first use-case it's needed, so added it.

**Review by yaronya — COMMENTED** [review]
Please also add a sentence to the docs, describing which TF value is required to reset the number of total environments. We can also specify that removing the value completely does the same (if it's not documented till now...)

**Comment by TomerHeber** [comment]
> Please also add a sentence to the docs, describing which TF value is required to reset the number of total environments. We can also specify that removing the value completely does the same (if it's not documented till now...)



> Please also add a sentence to the docs, describing which TF value is required to reset the number of total environments. We can also specify that removing the value completely does the same (if it's not documented till now...)

@yaronya - updated the text. Hopefully, now it's clearer.

**Review by yaronya — APPROVED** [review]

**Pull request merged**